File: Andy/src/level2/javascript/pairRemove.js
//프로그래머스 Level 2,짝지어 제거하기
function solution(s)
{
    function checkLast(arr,d) {
        if(arr.length == 0) return false;
        if(arr[arr.length-1].localeCompare(d) == 0) return true;
        return false;
    }
    
    var arr = [];    
    [...s].forEach( (d) => {
        if(checkLast(arr,d)) 
            arr.pop();
        else 
            arr.push(d);    
    });
    
    if(arr.length == 0) return 1;
    return 0;
}

File: Andy/src/level2/java/pairRemove.java
package level2.java;

import java.util.Stack;

//프로그래머스 Level 2,짝지어 제거하기
public class pairRemove {
    public int solution(String s)
    {
        Stack<String> stack = new Stack<>();        
        for(int i=0;i<s.length();i++) {
        	String tmp = String.valueOf(s.charAt(i));
        	if(stack.size()>0 && stack.peek().compareTo(tmp) == 0) {
        		stack.pop();
        	}
        	else {
        		stack.push(tmp);
        	}
        }
        
        return stack.size()>0?0:1;
    }
}
